third_party/move/move-compiler-v2/src/pipeline/explicit_drop.rs

```diff
@@ -0,0 +1,197 @@
+use super::{
+    livevar_analysis_processor::{LiveVarAnnotation, LiveVarInfoAtCodeOffset},
+    reference_safety_processor::{LifetimeAnnotation, LifetimeInfoAtCodeOffset},
+};
+use move_binary_format::file_format::CodeOffset;
+use move_model::{ast::TempIndex, model::FunctionEnv, ty::Type};
+use move_stackless_bytecode::{
+    function_target::{FunctionData, FunctionTarget},
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{AssignKind, AttrId, Bytecode, Operation},
+};
+use std::collections::BTreeSet;
+
+pub struct ExplicitDrop {}
+
+impl FunctionTargetProcessor for ExplicitDrop {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        fun_env: &FunctionEnv,
+        mut data: FunctionData,
+        _scc_opt: Option<&[FunctionEnv]>,
+    ) -> FunctionData {
+        if fun_env.is_native() {
+            return data;
+        }
+        let target = FunctionTarget::new(fun_env, &data);
+        let mut transformer = ExplicitDropTransformer::new(target);
+        transformer.transform();
+        data.code = transformer.transformed;
+        data.annotations.remove::<LiveVarAnnotation>();
```

Comment:
Do we want to completely remove this information? Or do we want to perhaps recompute changed information like after copy transformation in live var analysis?

We should clearly document the pre-requisite analysis as well as important side-effects (like removal of live var information).

Reply:
It's better to recompute the changed information so that there are no side effects. I'll do this to a later PR since we need to land this quickly.

Reply:
File an issue to record this.

Reply:
[#11378
](https://github.com/aptos-labs/aptos-core/issues/11378)